### PR TITLE
feat(HealthReport): create, update, and delete mutations

### LINKED ISSUE #17: create mutations for add, remove, and update HealthReport
create mutations for add, remove, and update HealthReport

### PR DESCRIPTION
**Pull request by austenm**
closes #17

### CODE REVIEW

File: src/HealthReport/resolvers.js [hunk 1]
```diff
@@ -20,12 +22,38 @@ const resolvers = {
...
+      healthReport.id === id),
     healthReportByPersonId: (_parent, { personId }) => healthReportsData.find((healthReport) =>
-      healthReport.personId === personId
-    ),
+      healthReport.personId === personId),
+  },
+  Mutation: {
+    createHealthReport: (_parent, { personId, bmi }) => {
+      const newHealthReport = {
+        id = uuid(),
```
Comment (MetaMatt1) [inline src/HealthReport/resolvers.js:32]: should be key value pairs inside an object. example: id: uuid(),

File: src/HealthReport/resolvers.js [hunk 2]
```diff
@@ -20,12 +22,38 @@ const resolvers = {
...
+        personId,
+        bmi,
+      };
+      healthReportsData.push(newHealthReport);
+      console.log(healthReportsData);
```
Comment (MetaMatt1) [inline src/HealthReport/resolvers.js:37]: clean out console.logs

File: src/HealthReport/resolvers.js [hunk 3]
```diff
@@ -20,12 +22,38 @@ const resolvers = {
...
+      return newHealthReport;
+    },
+    updateHealthReport: (_parent, { id, personId }) => {
+      const healthReportIndex = healthReportsData.findIndex((healthReport) =>
+        healthReport.personId === personId);
+      if (id) {
+        healthReportsData[healthReportIndex].id = id;
+      }
```
Comment (MetaMatt1) [inline src/HealthReport/resolvers.js:45]: we don't want to change the id of the healthReport.

Comment (MetaMatt1) [reply src/HealthReport/resolvers.js:45]: should be bmi not id

File: src/HealthReport/resolvers.js [hunk 4]
```diff
@@ -20,12 +22,38 @@ const resolvers = {
...
+      if (personId) {
+        healthReportsData[healthReportIndex].personId = personId;
+      }
```
Comment (MetaMatt1) [inline src/HealthReport/resolvers.js:48]: we don't want to change the personId

File: src/HealthReport/resolvers.js [hunk 5]
```diff
@@ -20,12 +22,38 @@ const resolvers = {
...
+      return healthReportsData[healthReportIndex];
+    },
+    deleteHealthReport: (_parent, { id }) => {
+      healthReportsData = healthReportsData.filter((healthReport) => healthReport.id !== id);
+      console.log(healthReportsData);
```
Comment (MetaMatt1) [inline src/HealthReport/resolvers.js:53]: clean out console.logs

File: src/HealthReport/typeDefs.js [hunk 6]
```diff
@@ -11,6 +11,12 @@ const typeDefs = gql`
     healthReportById(id: ID!): HealthReport
     healthReportByPersonId(personId: ID!): HealthReport
   }
+
+  type Mutation {
+    createHealthReport(personId: ID!, bmi: Float): HealthReport
+    updateHealthReport(id: ID, personId: ID!): HealthReport
```
Comment (MetaMatt1) [inline src/HealthReport/typeDefs.js:17]: remove id, add optional bmi

**Review by MetaMatt1 — CHANGES_REQUESTED** [review]

**Review by MetaMatt1 — APPROVED** [review]

**Review by mareklubomirski — APPROVED** [review]

**Pull request merged**